storage/watchable_store_bench_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+)
+
+func BenchmarkWatchableStoreUnsyncedCancel(b *testing.B) {
+	s := newWatchableStore(tmpPath)
+	defer cleanup(s, tmpPath)
+
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		if i < 10 {
+			continue
+		}
+		b.StopTimer()
+		keys := [][]byte{}
+		for j := 0; j < i; j++ {
```

Comment:
The testing method is incorrect IMO.

For N, it tests cancel time for {1, 2, 3, ...., N} watchers in the store. The bench time is an sum of these cases, so it cannot measure single case accurately. For example, if N is 10000, it doesn't tell me the cancel time in a big watcher map, but tells me the sum of cancel time in {1, 2, 3, .. N}-size map. Does this make sense to you?

Let us have more discussions before you move further.